osu.Game.Tournament/IO/TournamentStorage.cs

```diff
@@ -13,12 +15,15 @@ public class TournamentStorage : MigratableStorage
     {
         private const string default_tournament = "default";
         private readonly Storage storage;
+        private readonly Storage allTournaments;
```

Comment:
I'd make this a get-only delegating property:

```csharp
private Storage allTournamentsStorage => UnderlyingStorage;
```

The aliasing is nice for readability but the assignment is unnecessarily troublesome.

Reply:
Unfortunately doing that displays a different behaviour.

Currently when it's in the constructor as-is `UnderlyingStorage` is points to `tournaments/` because `ChangeTargetStorage` hasn't ran yet. 

Because the storage gets changed in the constructor, the suggested change lists the folders of `tournaments/default` rather than `tournaments`

Before: 

![image](https://user-images.githubusercontent.com/803255/101403168-1ebfae00-38d5-11eb-94a5-2cc7365296ca.png)

Suggested change: 
![image](https://user-images.githubusercontent.com/803255/101403236-3860f580-38d5-11eb-80c7-ca355c090609.png)

Reply:
Oh yikes, I missed the fact it's migrating itself inside the ctor... I guess this is more acceptable then, but I'd probably change all other accesses to `UnderlyingStorage` in the ctor to use `allTournaments` instead. Should hopefully make it clearer what's going on...?

Reply:
Yeah that will likely make it clearer. I will do that then instead